armi/physics/neutronics/crossSectionSettings.py

```diff
@@ -125,65 +181,136 @@ def isPregenerated(self):
         """True if this points to a pre-generated XS file."""
         return self.fileLocation is not None
 
+    def _validate(self):
+        """
+        Perform validation checks on the set of inputs.
+
+        This makes some design assumptions about the lattice
+        physics code (i.e., critical buckling search is not
+        a valid option for non-0D problems), so if different
+        behavior is desired then it is recommended to subclass
+        this and to make a new cross section setting.
```

Comment:
This is validation beyond what voluptuous does because it's extra logic, etc, right?

Reply:
Yeah

Reply:
Actually, the vol is performed only when an XSModelingOption is created in the `_load` method of XSSettingDef. It seems that the XSModelingOption object itself should be responsible for its our attribute/input validation and the vol. package only helps the user conform to the specific inputs when setting the `crossSectionSettings` in the case settings.

Reply:
Updated the docstring since it's not really validating design specifics here, just warning when things might be over defined. I agree that this stuff is design and application specific (see https://github.com/terrapower/armi/pull/289#discussion_r622737320)

Reply:
Since vol's job is to "validate data structures", regardless of where they are in their lifecycle, do you think we could use only vol for this if we refactor this system to traffic in more isolated data structures? 

Overall my biggest thought here is that we should add lots of info to the docstring of the module explaining what we think is wrong with this to help guide future framework users and developers because it all must be pretty surprising as is.